components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONRequestParser.java

```diff
@@ -105,113 +105,124 @@ public static RequestCtx parse(String jsonRequest) throws JsonParseException, Re
-                        if (jsonCategory.has(EntitlementEndpointConstants.ATTRIBUTE)) {
-                            if (jsonCategory.get(EntitlementEndpointConstants.ATTRIBUTE).isJsonArray()) {
-                                attributes = new HashSet<>();
-                                for (JsonElement jsonElement : jsonCategory.get(EntitlementEndpointConstants.ATTRIBUTE)
-                                        .getAsJsonArray()) {
-                                    attributes.add(jsonObjectToAttribute(jsonElement.getAsJsonObject()));
+    private static void jsonAttributeSeperator(Map.Entry<String, JsonElement> jsonAttribute, JsonObject jsonCategory, Set<Attributes> categories) throws RequestParseException, UnknownIdentifierException {
+
+        Node content = null;
+        URI category = null;
+        Set<Attribute> attributes = null;
+        String id = null;
+        switch (jsonAttribute.getKey()) {
+
+            //For a custom Category
+            //Or a Category with long identifier
+            case EntitlementEndpointConstants.CATEGORY_DEFAULT:
+                if (jsonCategory.has(EntitlementEndpointConstants.CATEGORY_ID)) {
+                    category = stringCateogryToURI(jsonCategory
+                            .get(EntitlementEndpointConstants.CATEGORY_ID)
+                            .getAsString());
+                }
+            case EntitlementEndpointConstants.CATEGORY_RESOURCE:
+            case EntitlementEndpointConstants.CATEGORY_ACTION:
+            case EntitlementEndpointConstants.CATEGORY_ENVIRONMENT:
+            case EntitlementEndpointConstants.CATEGORY_ACCESS_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_RECIPIENT_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_INTERMEDIARY_SUBJECT:
+            case EntitlementEndpointConstants.CATEGORY_CODEBASE:
+            case EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE:
 
-                                }
-                            }
+                if (category == null) {
+                    category = stringCateogryToURI(jsonAttribute.getKey());
+                }
+                if (jsonCategory.has(EntitlementEndpointConstants.ID)) {
+                    id = jsonCategory.get(EntitlementEndpointConstants.ID).getAsString();
+                }
+                if (jsonCategory.has(EntitlementEndpointConstants.CONTENT)) {
+                    ByteArrayInputStream inputStream;
+                    DocumentBuilderFactory dbf;
+                    Document doc = null;
+
+                    String xmlContent = stringContentToXMLContent(jsonCategory
+                            .get(EntitlementEndpointConstants.CONTENT)
+                            .getAsString());
+                    inputStream = new ByteArrayInputStream(xmlContent.getBytes());
+                    dbf = IdentityUtil.getSecuredDocumentBuilderFactory();
+                    dbf.setNamespaceAware(true);
+
+                    try {
```

Comment:
Shall we use try-with-resources statements here?
It will ensure that each resource is closed at the end of the statement without us explicitly doing that.